Applications/Spire/Include/Spire/KeyBindings/KeyBindingsWindow.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef SPIRE_KEY_BINDINGS_WINDOW_HPP
+#define SPIRE_KEY_BINDINGS_WINDOW_HPP
+#include "Spire/KeyBindings/KeyBindings.hpp"
+#include "Spire/KeyBindings/OrderTasksPage.hpp"
+#include "Spire/Ui/ComboBox.hpp"
+#include "Spire/Ui/Window.hpp"
+
+namespace Spire {
+
+  /** Display the key bindings window. */
+  class KeyBindingsWindow : public Window {
+    public:
+
+      /**
+       * Signals the submission of key bindings of order tasks.
+       * @param submission The order tasks key bindings to be submitted.
+       */
+      using SubmitSignal = Signal<void (
+        const boost::optional<std::shared_ptr<ListModel<OrderTask>>>&
+          order_tasks_submission)>;
+
+      /** Stores the various models used to build a KeyBindingsWindow. */
+      struct KeyBindingsModels {
```

Comment:
Remove this struct, it doesn't look like it's used for anything other than the constructor.

Reply:
It concerns me that there are too many parameters in the constructor. Later, four models will be added into the constructor, such as cancel orders model, default cancel orders model, interactions model and default interactions model.

Reply:
In that case you can create a ```KeyBindingsModel``` class, look at the ```blotter``` branch and look at the ```BlotterModel``` for an example.

Reply:
https://github.com/spiretrading/nexus/blob/blotter/Applications/Spire/Include/Spire/Blotter/BlotterModel.hpp